### PR TITLE
Remove legacy prefix when filtering workshop by subject

### PR DESCRIPTION
**Pull request by hacodeorg**
In https://github.com/code-dot-org/code-dot-org/pull/30203, I added `[Legacy]` prefix to workshop subject names. This PR removes that prefix, keeps subject names unchanged, and change only subject labels in drop-down list. This allows workshop filtering by subject name to work properly.

### CODE REVIEW

**Review by islemaster — COMMENTED** [review]
This doesn't feel great - it's another case where we need to separate this display name from a key value.  Is there another approach to this?

Alternatively, is this urgent?  What bug does this fix?

**Comment by hacodeorg** [comment]
The bug is when you filter workshops using legacy subject names, it returns empty result. The reason is selected value in drop-down list is prefixed by _`[Legacy]`_ while the real (legacy) subject name is not.

This may not be super urgent, unless facilitators/regional partner wants to review ended workshop results.

Alternative fix could be to separate value and label displayed in drop-down list. Only add prefix to the labels but not the values.

> This doesn't feel great - it's another case where we need to separate this display name from a key value. Is there another approach to this?
> 
> Alternatively, is this urgent? What bug does this fix?

**Comment by islemaster** [comment]
> separate value and label displayed in drop-down list.

That solution feels better to me - especially if we think about moving toward making this UI translatable.

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx [hunk 1]
```diff
@@ -172,7 +172,7 @@ export class WorkshopFilter extends React.Component {
   };
 
   handleSubjectChange = selected => {
-    const subject = selected ? selected.value : null;
+    let subject = selected ? selected.value : null;
```
Comment (hacodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx]: Will revert this line

**Review by breville — APPROVED** [review]

**Comment by breville** [comment]
Is there any test coverage for this UI?

**Comment by hacodeorg** [comment]
I don't see an UI test for this view. I can add unit test for the new functions.
> Is there any test coverage for this UI?

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx [hunk 2]
```diff
@@ -313,23 +313,34 @@ export class WorkshopFilter extends React.Component {
...
+        (result[course] = subjects[course].map(subject => ({
+          value: subject,
+          label: prefix + subject
+        })))
+    );
+
+    return result;
+  }
+
+  static concatSubjetArrays(objValue, srcValue) {
```
Comment (islemaster) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx]: Subjet -> Subject

**Review by islemaster — APPROVED** [review]
Very nice!

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx (hunk 2 above)
Comment (hacodeorg) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx]: hmm, I should enable spelling check on Rubymine

**Pull request merged**